### PR TITLE
Changed SaveAs[Async](string) to CopyTo[Async](Stream)

### PR DESCRIPTION
**Pull request by khellang**
#527

### CODE REVIEW

File: src/Microsoft.AspNet.Http/Features/FormFile.cs [hunk 1]
```diff
@@ -74,28 +74,20 @@ public Stream OpenReadStream()
...
-        public void SaveAs(string path)
+        /// <param name="target">The stream to save the file to.</param>
+        public void Save(Stream target)
         {
-            using (var fileStream = File.Create(path, DefaultBufferSize))
-            {
-                var inputStream = OpenReadStream();
-                inputStream.CopyTo(fileStream, DefaultBufferSize);
-            }
+            OpenReadStream().CopyTo(target, DefaultBufferSize);
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Http/Features/FormFile.cs]: Shouldn't we be disposing our stream? If we don't who does?

Comment (khellang) [inline src/Microsoft.AspNet.Http/Features/FormFile.cs]: Yeah, I've seen it being disposed other places...

**Comment by muratg** [comment]
@Tratcher Could you have a look?

**Comment by Tratcher** [comment]
@lodejard @davidfowl What's the point of having IFormFile.CopyTo(Stream) when you already have IFormFile.OpenReadStream().CopyTo(Stream)?

Should we just remove SaveAs(string)?

**Comment by khellang** [comment]
> What's the point of having IFormFile.CopyTo(Stream) when you already have IFormFile.OpenReadStream().CopyTo(Stream)?

Good point :stuck_out_tongue_closed_eyes::+1:

**Comment by Tratcher** [comment]
Hmm I guess it would actually be something like:
```
IFormFile.CopyTo(Stream output)
{
 using (var readStream = OpenReadStream())
 {
  reasStream.CopyTo(output);
 }
}
```
This might be worth it if it was actually important to dispose the readStream, but so far it isn't.

**Comment by Tratcher** [comment]
Rebased and merged.

**Pull request closed without merging**